src/shogun/multiclass/CoverTreeKNNSolver.cpp

```diff
@@ -58,7 +57,7 @@ CMulticlassLabels* CCoverTreeKNNSolver::classify_objects(CDistance* distance, co
 	v_array< v_array< CJLCoverTreePoint > > res;
 	k_nearest_neighbor(top, top_query, res, m_k);
 
-#ifdef DEBUG_KNN
+#ifdef SG_DEBUG
```

Comment:
@MikeLing noup i didn't mean that at all.... and yes with your code that's the problem what you have just explained after buuuuuut that's why i said,
`you wanna check the current log_level on runtime`, namely:
```
if (io->get_loglevel() == MSG_DEBUG) {
/* debugging code */
}
```